GitUI/UserControls/RevisionGrid/Columns/MessageColumnProvider.cs

```diff
@@ -144,15 +144,12 @@ public override bool TryGetToolTip(DataGridViewCellMouseEventArgs e, GitRevision
                 return true;
             }
 
-            if (revision.IsArtificial)
+            // Artificial
+            var changeCount = _grid.GetChangeCount(revision.ObjectId);
+            if (changeCount is not null && AppSettings.ShowGitStatusForArtificialCommits)
```

Comment:
With the explicit type, the code becomes readable (even without the too short comment) for others than you, too.
I don't see a need to check for `AppSettings.ShowGitStatusForArtificialCommits` here. It would change the behavior at least for this part of code.

```suggestion
            ArtificialCommitChangeCount changeCount = _grid.GetChangeCount(revision.ObjectId);
            if (changeCount is not null)
```

Reply:
Will add the type.

For the settings check, this was primarily to simplify review, I looked at this some time without understanding why this worked.
Secondary to protect from accidental changes. The status is (currently) null if the setting is disabled. If someone wanted the tooltip to include "updating" or so in this scenario, it would require the settings check.
But I can remove the check. Maybe replace with a comment (which is adding to the 'codemass' not what I prefer.
`// if AppSettings.ShowGitStatusForArtificialCommits is not set, changeCount is always null`

Reply:
ok